src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -1923,108 +1942,176 @@ export class WASMExpressionGen {
         rightValue?: SemanticsValue,
     ) {
         const owner = value.owner as VarValue;
-        const meta = owner.shape!.meta;
-        const member = meta.members[value.index];
+        const shapeMeta = owner.shape!.meta;
+        const shapeMember = shapeMeta.members[value.index];
         const ownerType = owner.type as ObjectType;
         let targetValue = value.value!;
         if (rightValue) {
             targetValue = rightValue;
         }
         const typeMeta = ownerType.meta;
-        return this.setInstField(
+        const typeMember = typeMeta.findMember(
+            shapeMember.name,
+        ) as MemberDescription;
+        return this.setInstMember(
             this.wasmExprGen(owner),
             targetValue,
             ownerType,
             typeMeta,
-            member,
+            typeMember,
         );
     }
 
-    private setInstField(
+    private setInstMember(
         thisRef: binaryen.ExpressionRef,
         targetValue: SemanticsValue,
         ownerType: ObjectType,
-        meta: ObjectDescription,
-        member: MemberDescription,
+        typeMeta: ObjectDescription,
```

Comment:
Can we summarize a document about "shapeMeta" and "typeMeta"?

Reply:
Good idea, maybe we need a document to explain the used nouns.